ecommerce/views_test.py

```diff
@@ -693,6 +720,29 @@ def test_patch_basket_update_active_inactive_product(basket_client, basket_and_c
     assert resp.status_code == status.HTTP_200_OK
 
 
+def test_patch_basket_update_active_inactive_product_readable_id(
+    basket_client, basket_and_coupons
+):
+    """ Test that inactive product id is rejected with no changes to basket but not the active ones. """
+    product = ProductVersionFactory.create().product
+    product.is_active = False
+    product.save()
+    readable_id = get_readable_id(product.content_object)
+    data = {"items": [{"product_id": readable_id}]}
+    resp = basket_client.patch(reverse("basket_api"), type="json", data=data)
+    assert resp.status_code == status.HTTP_400_BAD_REQUEST
+    resp_data = resp.json()
+    assert (
+        "Product id {product_id} is not active".format(product_id=readable_id)
+        in resp_data["errors"]["items"]
+    )
+
+    product.is_active = True
+    product.save()
+    resp = basket_client.patch(reverse("basket_api"), type="json", data=data)
+    assert resp.status_code == status.HTTP_200_OK
```

Comment:
These 4 lines aren't necessary. They are already tested in `test_patch_basket_new_item_with_readable_id`